feat: add readme (#3)
topos-network/go-topos-sequencer-client: README.md

<div id="top"></div>
<!-- PROJECT LOGO -->
<br />
<div align="center">

  <img src="./.github/assets/topos_logo.png#gh-light-mode-only" alt="Logo" width="200">
  <img src="./.github/assets/topos_logo_dark.png#gh-dark-mode-only" alt="Logo" width="200">

<br />

<p align="center">
<em>go-topos-sequencer-client</em> is an early proof of concept client for Golang GRPC bidirectional communication.
</p>

<br />

</div>

## Background

> The Topos sequencer produces and signs certificates using FROST signatures (FROST: Flexible Round-Optimised Schnorr Threshold signatures). It requires communication between sequencers of the same subnet to generate distributed keys and sign certificates with joint effort using threshold signature protocol. One way to implement this protocol is to reuse the existing p2p network that the subnet nodes maintain. `go-topos-sequencer-client` will eventually be used for bidirectional communication between subnet nodes and their Topos sequencers to generate distributed keys and sign certificates reusing the subnet p2p network.

Reviewer: ```suggestion
The Topos Sequencer produces and signs certificates using [ICE-FROST](https://eprint.iacr.org/2021/1658.pdf) signatures. It requires communication between sequencers of the same subnet to proceed to the DKG to generate the group key, and sign certificates with joint effort using the threshold signature protocol.

One way to implement this protocol is to reuse the existing p2p network that the subnet nodes maintain. `go-topos-sequencer-client` will eventually be used for bidirectional communication between subnet nodes and their Topos sequencers to generate the group key and sign certificates by leveraging the subnet p2p network.
```

<div id="top"></div>
<!-- PROJECT LOGO -->
<br />
<div align="center">

  <img src="./.github/assets/topos_logo.png#gh-light-mode-only" alt="Logo" width="200">
  <img src="./.github/assets/topos_logo_dark.png#gh-dark-mode-only" alt="Logo" width="200">

<br />

<p align="center">

> go-topos-sequencer-client is simple early proof of concept client for Golang GRPC bidirectional communication.

Reviewer 2: ```suggestion
`go-topos-sequencer-client` is an early proof of concept client for Golang GRPC bidirectional communication.
```

</p>

<br />

</div>

## Background

> Topos sequencer is responsible to produce and sign certificates using FROST signatures (FROST: Flexible Round-Optimised Schnorr Threshold signatures). It requires communication between sequencers of the same subnet to generate distributed keys and sign certificates with joint effort using threshold signature protocol. One way to implement this protocol is to reuse already existing p2p network that subnet nodes are maintaining. `go-topos-sequencer-client` will be eventually used for bidirectional communication between subnet node and Topos sequencer for the purpose of generating distributed keys and signing certificates reusing subnet p2p network.

Reviewer 2: ```suggestion
The Topos sequencer produces and signs certificates using FROST signatures (FROST: Flexible Round-Optimised Schnorr Threshold signatures). It requires communication between sequencers of the same subnet to generate distributed keys and sign certificates with joint effort using threshold signature protocol. One way to implement this protocol is to reuse the existing p2p network that the subnet nodes maintain. `go-topos-sequencer-client` will eventually be used for bidirectional communication between subnet nodes and their Topos sequencers to generate distributed keys and sign certificates reusing the subnet p2p network.
```
